File: BubbleSort.java
package BubbleSort;

public class BubbleSort
{
    public static void main(String[] args) {
        int [] unsortedArr = {10, -1, 34, 89, 50, -7, -189};
        int [] sortedArr = bubbleSort (unsortedArr);

        for (int i = 0; i < sortedArr.length; i++){
            System.out.println (sortedArr [i]);
        }
    }

    public static int [] bubbleSort (int [] unsortedArr){
        int unsortedIndex = unsortedArr.length - 1;
        while (unsortedIndex != 0){
            for (int i = 0; i < unsortedIndex; i++){
                if (unsortedArr [i] > unsortedArr [i + 1]){
                    int temp = unsortedArr [i];
                    unsortedArr [i] = unsortedArr [i + 1];
                    unsortedArr [i + 1] = temp;
                }
            }
            unsortedIndex --;
        }

        return unsortedArr;
    }
}

File: BubbleSort.py
# -*- coding: utf-8 -*-

def main ():
    
    unsortedArr = [10, -1, 34, 89, 50, -7, -189]
    sortedArr = bubbleSort (unsortedArr)
    print (sortedArr)
    
    assert (sortedArr == sorted (unsortedArr))
    return
    
def bubbleSort (unsortedArr):
    unsortedIndex = len (unsortedArr) - 1
    while (unsortedIndex > 0):
        for i in range (unsortedIndex):
            if unsortedArr [i] > unsortedArr [i + 1]:
                temp = unsortedArr [i]
                unsortedArr [i] = unsortedArr [i + 1]
                unsortedArr [i + 1] = temp
        unsortedIndex -= 1
        
    return unsortedArr
    
main ()